contracts/SponsorFaucet.sol

```diff
@@ -80,6 +80,7 @@ contract SponsorFaucet is
             largeBounds.collateral_bound <= largeBounds.collateral_total_limit,
             "bound higher than limit"
         );
+        large_contracts[large] = true;
```

Comment:
Why not handle the case in function `getBounds`?

Reply:
add check in **isLargeContract** to handle this